### ISSUE TITLE
Nothing returned by handler for mockQueryRecord when mocking ember-simple-auth currentUser

### ISSUE THREAD

**Issue by BryanCrotaz** [danielspaniel/ember-data-factory-guy#415]
`ember-data-factory-guy@3.9.5`

I think I've followed the instructions.
I ran `ember g factory user` then created the below test:

Result:
```

Source: | Error: 
Pretender intercepted GET /api/v1/users/current but encountered an error: 
Nothing returned by handler for /api/v1/users/current. Remember to `return [status, headers, body];` in your route handler.     
at _handleRequest (webpack://__ember_auto_import__/./node_modules/pretender/dist/pretender.es.js?:398:17)
at Pretender.handleRequest (webpack://__ember_auto_import__/./node_modules/pretender/dist/pretender.es.js?:419:11)
at FakeRequest.send (webpack://__ember_auto_import__/./node_modules/pretender/dist/pretender.es.js?:186:21)
at eval (webpack://__ember_auto_import__/./node_modules/whatwg-fetch/fetch.js?:524:9)
at new Promise (<anonymous>)     at fetch (webpack://__ember_auto_import__/./node_modules/whatwg-fetch/fetch.js?:461:10)
at self.default (http://localhost:7357/assets/vendor-ffef3f548e3661a0cddcf91a565f3c36.js:97736:27)
at Class._fetchRequest (http://localhost:7357/assets/vendor-ffef3f548e3661a0cddcf91a565f3c36.js:109364:16)
at Class.ajax (http://localhost:7357/assets/vendor-ffef3f548e3661a0cddcf91a565f3c36.js:109311:21)
at Class.queryRecord (http://localhost:7357/assets/vendor-ffef3f548e3661a0cddcf91a565f3c36.js:108943:19)
```

```js
// test.js
import { module, test } from 'qunit';
import { visit, click } from '@ember/test-helpers';
import { setupApplicationTest } from 'ember-qunit';
import { setupFactoryGuy, make, mockQueryRecord, mockFindRecord } from 'ember-data-factory-guy';
import { authenticateSession } from 'ember-simple-auth/test-support';

module('Acceptance | edit player shrink settings', function(hooks) {
  setupApplicationTest(hooks);
  setupFactoryGuy(hooks);

  test('enable led shrink shows save/cancel buttons and cancel hides them', async function(assert) {
    let player = make('player');
    let mockPlayer = mockFindRecord('player').returns({model: player});

    let user = make('user');
    mockQueryRecord('user', {current: true}).returns({model: user});

    await authenticateSession({
      authToken: '12345'
    });  // calls await this.store.queryRecord('user', {current: true}); under the hood

    await visit('/players/'+mockPlayer.id);

    await this.pauseTest();
    await click('[data-test-shrink-enabled]');
    // TODO rest of test
  });
});
```
```js
// user/adapter.js
import ApplicationAdapter from 'sc-pp-frontend/adapters/application';
import config from '../config/environment';

export default ApplicationAdapter.extend({
  urlForQueryRecord(query) {
    if (query.current) {
      // get the current user
      delete query.current;
      return `${this._super(...arguments)}/current`;
    }
    return this._super(...arguments);
  }
});
```

**Comment by danielspaniel** [comment]
first off .. instead of this >>
```let mockPlayer = mockFindRecord('player').returns({model: player});```
do this:
```let mockPlayer = mockFindRecord(player);```


as to this
``` let user = make('user');
    mockQueryRecord('user', {current: true}).returns({model: user});
```

not sure what you doing wrong.  you might try using mock instead of mockQueryRecord 

 https://github.com/danielspaniel/ember-data-factory-guy#mock

 but I can't do more than this without actually debugging your code step by step to see why your url not capturing

**Comment by BryanCrotaz** [comment]
I'm debugging step by step and I've found the handler isn't being matched.

**Comment by danielspaniel** [comment]
Right, so what can I do. I suggested using mock  and setting the url  exactly ( depending on what the handleRequest url is getting. 

 you can also check the mock.getURL() function to
 see what it is mocking ( mockQueryRequest ) 

 other than that .. you gotta track it down.   if its a bug then that is fine but for now I don't know yet

**Comment by BryanCrotaz** [comment]
what should the handler look like with the test code I wrote? Should handler.handler.handlers be an array of one object?

**Comment by BryanCrotaz** [comment]
Aha. Found the problem.

```
    let mockQuery = mockQueryRecord('user', {current: true}).returns({model: user});
    console.log(mockQuery.getUrl());
    //logs /api/v1/users -- should be /api/v1/users/current
```

**Comment by danielspaniel** [comment]
there you go .. so simple

**Comment by BryanCrotaz** [comment]
But the mock is picking up the correct url from my adapter. Can't currently see why it's going wrong

**Comment by danielspaniel** [comment]
i don't know either frankly .. sorry can't be more help.

**Comment by BryanCrotaz** [comment]
To check, the handler should pick up the url from the adapter, and then that url should be returned by `getUrl`?

If that's the correct behaviour I can try to create a failing test with my setup and submit a PR.

**Comment by danielspaniel** [comment]
i am not 100% sure actually. but factory guy has boat load of tests .. why don't you check there ?  if you submit PR with failing test, you will have to fix it yourself as well. Not have time for fixing up things now.

**Comment by BryanCrotaz** [comment]
Works with ember-data 3.9, fails spectacularly with 437 test failures with ember-data 3.18 (which is what I'm using). Investigating.

**Comment by danielspaniel** [comment]
yeah , i have not updated factory guy for the latest ember  , and since I am not using ember anymore, I am not really into doing that. hope that is not a killer for you trying to fix though

**Comment by BryanCrotaz** [comment]
Found the problem. Submitting a PR.

**Comment by BryanCrotaz** [comment]
The failing tests with latest `ember-data` appear to be a problem with `ember-data-model-fragments`.

**Comment by danielspaniel** [comment]
model fragments have not updated in a while, so ?  not much we can do i reckon. but thanks for heads up.

**Comment by BryanCrotaz** [comment]
There is a fix coming in model fragments, a couple of weeks away they say

**Comment by danielspaniel** [comment]
what fails in master ? confused, since I just merged your PR since all tests passed in master with it

**Comment by BryanCrotaz** [comment]
Look at the 1 line code change. There’s a comment with it explaining it. The test I’ve added fails in master but doesn’t fail with the code fix I’ve also supplied.

**Comment by BryanCrotaz** [comment]
Fixed by #416
